### PR TITLE
Publish facts gathering requested

### PR DESCRIPTION
**Pull request by fabriziosestito**
Publish FactsGatheringRequested event

### CODE REVIEW

**Review by rtorrero — APPROVED** [review]
LGTM

File: lib/wanda/messaging/adapters/amqp.ex [hunk 1]
```diff
@@ -8,7 +8,13 @@ defmodule Wanda.Messaging.Adapters.AMQP do
   alias Wanda.Messaging.Adapters.AMQP.Publisher
 
   @impl true
+
+  # FIXME: fix Trento.Contracts.to_event/2
```
[the remark this answers is not in the document]
Comment (fabriziosestito) [reply lib/wanda/messaging/adapters/amqp.ex:12]: the dialyzer complains about a function that will not return in the contracts repo, which depends on the `protobuf` library. Unfortunately, this bubbles up here, but the dialyzer is wrong (wrong spec in the probouf code?, not sure)

File: lib/wanda/messaging/adapters/amqp.ex [hunk 2]
```diff
@@ -8,7 +8,13 @@ defmodule Wanda.Messaging.Adapters.AMQP do
...
+  @dialyzer {:nowarn_function, publish: 2}
+
   def publish(routing_key, message) do
-    Publisher.publish_message(message, routing_key)
+    message
+    |> Trento.Contracts.to_event(source: "wanda")
```
Comment (arbulu89) [inline lib/wanda/messaging/adapters/amqp.ex:17]: Maybe the source could be `github.com/trento-project/wanda`

**Review by arbulu89 — APPROVED** [review]
Greaaaat!

File: lib/wanda/messaging/adapters/amqp.ex (hunk 2 above)
Comment (fabriziosestito) [reply lib/wanda/messaging/adapters/amqp.ex:17]: good one, we could take this as inspiration: https://github.com/googleapis/google-cloudevents/blob/main/examples/structured/pubsub/MessagePublishedData.json

**Pull request merged**